filter/filters.go

```diff
@@ -35,7 +35,7 @@ func (Any) ValuesRange() *Range {
 }
 
 // EQ is a shorthand for Equal.
-func EQ(v values.Value) SortableFilter {
+func EQ(v values.Value) Equal {
```

Comment:
You probably noticed that you were unable to change return type to a specific one in function like `Prefix`. There's exactly the reason why it must be kept an interface: this implementation may do additional checks on parameters and decide which filter is the best suited to achieve the goal.

`EQ` might be a good example of this as well. For example, what should it do if user passes `float64(NaN)`? It could actually just optimize it out to `false` (since in usually `NaN != NaN`).

So I would prefer to keep return types as `SortableFilter` in all functions like these and only change to specific types if someone raises an issue with the actual problem about it.

Reply:
I have dropped the concerned commit 😉